components/cluster/command/prune.go

```diff
@@ -73,7 +73,7 @@ func destroyTombstoneIfNeed(clusterName string, metadata *spec.ClusterMeta, opt
 		return perrs.AddStack(err)
 	}
 
-	nodes, err := operator.DestroyTombstone(ctx, topo, true /* returnNodesOnly */, opt, tlsCfg)
+	nodes, err := operator.DestroyTombstone(ctx, topo, true /* returnNodesOnly */, opt, tlsCfg, ctx.PublicKeyPath)
```

Comment:
`ctx` is passed to `DestroyTombstone()` so I don't think it's necessary to pass `ctx.PublicKeyPath` again.

Reply:
The first argument inside `DestoryTombestone()` is `ExecutorGetter` interface: https://github.com/pingcap/tiup/blob/cc41ad5af87bde821ecde48d63dd7c1c40d72ea2/pkg/cluster/operation/destroy.go#L414-L420  So I need to cast it into `task.Context`, but this results in `import cycle` error. Please let me know if I've misssed something.

Reply:
Maybe we can define another interface for the first param of `DestroyTombstone`:
```golang
type SSHExecutorGetter interface {
    ExecutorGetter
    GetSSHKeySet() (string, string) // return both priv and public key
}
```

Reply:
@lucklove Or should we extend the original `ExecutorGetter` interface with:

```go
type ExecutorGetter interface {
	Get(host string) (e executor.Executor)
	GetSSHKeySet() (string, string)
}
```

Reply:
@lucklove I did it this way first, if you have any question, please let me know.

Reply:
> @lucklove Or should we extend the original `ExecutorGetter` interface with:
> 
> ```go
> type ExecutorGetter interface {
> 	Get(host string) (e executor.Executor)
> 	GetSSHKeySet() (string, string)
> }
> ```

It's OK at this time. In the future there may be some executors that don't relay on ssh, at that time we can refactor this.